clients/py/v3io_frames/pbutils.py

```diff
@@ -226,6 +262,91 @@ def series2col(s, name):
     return fpb.Column(**kw)
 
 
+def insert_nulls_based_on_bitmask(df, bitmask):
+    # if there are no Null values at all, skip
+    if len(bitmask) == 0:
+        return df
+    i = 0
+    casted_columns = {}
+    for key in df.index:
+        for col_name in bitmask[i].nullColumns:
+            # boolean columns should be converted to `object` to be able to represent None.
+            if df[col_name].dtype == np.bool and col_name not in casted_columns:
+                casted_columns[col_name] = True
+                df[col_name] = df[col_name].astype(object)
+            df.at[key, col_name] = None
+        i = i+1
+    return df
+
+
+def normalize_df(df, schema):
+    """
+        This function converts all 'Null' values to the according default values based on the column type,
+        and creates a bitmask column to specify where are the null values
+    :param schema: dictionary specifying the real type of every column
+    :param df:
+    :return:
+    """
+    bitmask = []
+    for index, row in df.iterrows():
+        current_bitmask = []
+        for col_name in df.columns:
+            if pd.isnull(row[col_name]):
+                df.at[index, col_name] = get_empty_value_by_type(schema[col_name])
+                current_bitmask.append(col_name)
+        bitmask.append({'nullColumns': current_bitmask})
```

Comment:
Just keep a list of lists?

Reply:
Can't do. i need an "object" named `nullColumns` otherwise protobuf is unable to create the Frame object

Reply:
inconsistent representation of numeric data types. Uninitialised integers will be initialized with type's default value (0) while floats with NaN